File: 1381. Design a Stack With Increment Operation/solution.java
class CustomStack {
    private int[] nums;
    private int mxSize;
    private int topIndex;

    public CustomStack(int maxSize) {
        this.mxSize = maxSize;
        this.nums = new int[mxSize];
        this.topIndex = -1;
    }
    
    public void push(int x) {
        if (topIndex == mxSize - 1) return;
        nums[++topIndex] = x;
    }
    
    public int pop() {
        if (topIndex == -1) return -1;
        return nums[topIndex--];
    }
    
    public void increment(int k, int val) {
        for (int i = 0; i < k && i <= topIndex; i++) {
            nums[i] += val;
        }
    }
}

File: 1381. Design a Stack With Increment Operation/solution.py
class CustomStack:
    def __init__(self, maxSize: int):
        self.nums = []
        self.mxSize = maxSize

    def push(self, x: int) -> None:
        if len(self.nums) == self.mxSize:
            return
        self.nums.append(x)

    def pop(self) -> int:
        if not self.nums:
            return -1
        return self.nums.pop()

    def increment(self, k: int, val: int) -> None:
        for i in range(min(k, len(self.nums))):
            self.nums[i] += val